CONTRIBUTING.md

```diff
@@ -0,0 +1,92 @@
+# Contributing
+
+So you want to contribute to Requests? Fantastic! There are a few rules you'll need to follow for all contributions.
+
+(There are always exceptions to these rules. :) )
+
+## Process
+
+1. Ideally, start with an issue to check the need for a PR. It's possible that a feature may be rejected at an early stage, and it's better to find out before you write the code.
+2. Write the code. Small, atomic commits are preferred. Explain the motivation behind the change when needed.
+3. File a PR. If it isn't ready for merge yet, note that in the description. If your PR closes an existing issue, add "fixes #xxx" to the message, so that the issue will be closed when the PR is merged.
+4. If needed, iterate on the code until it is ready. This includes adding unit tests. When you're ready, comment that the PR is complete.
+5. A committer will review your code and offer you feedback.
+6. Update with the feedback as necessary.
+7. PR will be merged.
+
+Notes:
+
+* All code needs to go through peer review. Committers may not merge their own PR.
+* PRs should **never be squashed or rebased**, even when merging. Keeping the history is important for tracking motivation behind changes later.
+
+## Compatibility
+
+All code in Requests must be compatible with PHP 5.6.
+Requests is used in WordPress, and hence is tied to [its compatibility][wp-requirements].
+
+Requests is also bound to not break backwards compatibility.
+In semantic versioning terms, this means we will be in the 2.x release cycle for the forseeable future.
```

Comment:
The dash in `BC-break` is not consistent with `backward-compatibility breaks` (no dash between BC and break).

I'd suggest:
- introduce the acronym at first usage: `backward-compatibility (BC) break`
- use `BC break` instead of `BC-break`